Code/Framework/AzCore/AzCore/Preprocessor/Enum.h

```diff
@@ -69,44 +69,58 @@
+#define GET_ENUM_OPTION_INITIALIZER_0()
 #define GET_ENUM_OPTION_INITIALIZER_1(_name) _name,
 #define GET_ENUM_OPTION_INITIALIZER_2(_name, _initializer) _name = _initializer,
 #define GET_ENUM_OPTION_INITIALIZER(_name) AZ_MACRO_CALL_INDEX(GET_ENUM_OPTION_INITIALIZER_, _name)
 
 #define INIT_ENUM_STRING_PAIR_IMPL(_enumname, _optionname) { _enumname::_optionname, AZ_STRINGIZE(_optionname) },
-#define INIT_ENUM_STRING_PAIR(_enumname, _optionname) INIT_ENUM_STRING_PAIR_IMPL(_enumname, GET_ENUM_OPTION_NAME(_optionname))
+#define INIT_ENUM_STRING_PAIR(_enumname, _optionname) AZ_ENUM_CALL( \
+    AZ_JOIN(INIT_ENUM_STRING_PAIR_, AZ_VA_NUM_ARGS(AZ_UNWRAP(_optionname))), \
+    _enumname, _optionname)
+// Wrapper macro used to delegate to the enum option name to string macro
+// If a line only contains a comma it would expand to an empty argument
+#define INIT_ENUM_STRING_PAIR_0(_enumname, _optionname)
+#define INIT_ENUM_STRING_PAIR_1(_enumname, _optionname) INIT_ENUM_STRING_PAIR_IMPL(_enumname, GET_ENUM_OPTION_NAME(_optionname))
+#define INIT_ENUM_STRING_PAIR_2(_enumname, _optionname) INIT_ENUM_STRING_PAIR_1(_enumname, _optionname)
 
 #define MAKE_REFLECTABLE_ENUM_UNSCOPED(EnumTypeName, ...) MAKE_REFLECTABLE_ENUM_(, EnumTypeName,, __VA_ARGS__)
 #define MAKE_REFLECTABLE_ENUM_UNSCOPED_WITH_UNDERLYING_TYPE(EnumTypeName, EnumUnderlyingType, ...) MAKE_REFLECTABLE_ENUM_(, EnumTypeName, : EnumUnderlyingType, __VA_ARGS__)
 #define MAKE_REFLECTABLE_ENUM_SCOPED(EnumTypeName, ...) MAKE_REFLECTABLE_ENUM_(class, EnumTypeName,, __VA_ARGS__)
 #define MAKE_REFLECTABLE_ENUM_SCOPED_WITH_UNDERLYING_TYPE(EnumTypeName, EnumUnderlyingType, ...)   MAKE_REFLECTABLE_ENUM_(class, EnumTypeName, : EnumUnderlyingType, __VA_ARGS__)
 #define MAKE_REFLECTABLE_ENUM_(SCOPE_QUAL, EnumTypeName, EnumUnderlyingType, ...) \
     MAKE_REFLECTABLE_ENUM_ARGS(SCOPE_QUAL, EnumTypeName \
-    , EnumUnderlyingType, AZ_VA_NUM_ARGS(__VA_ARGS__), __VA_ARGS__)
+    , EnumUnderlyingType, __VA_ARGS__)
 
-#define MAKE_REFLECTABLE_ENUM_ARGS(SCOPE_QUAL, EnumTypeName, EnumUnderlyingType, EnumTypeCount, ...) \
+#define MAKE_REFLECTABLE_ENUM_ARGS(SCOPE_QUAL, EnumTypeName, EnumUnderlyingType, ...) \
 inline namespace AZ_JOIN(EnumTypeName, Namespace) \
 { \
     enum SCOPE_QUAL EnumTypeName EnumUnderlyingType \
     { \
         AZ_FOR_EACH(GET_ENUM_OPTION_INITIALIZER, __VA_ARGS__) \
     };\
-    inline constexpr size_t AZ_JOIN(EnumTypeName, Count) = EnumTypeCount; \
     struct AZ_JOIN(EnumTypeName, EnumeratorValueAndString) \
     { \
         EnumTypeName m_value; \
         AZStd::string_view m_string; \
     }; \
-    constexpr AZStd::array<AZ_JOIN(EnumTypeName, EnumeratorValueAndString), AZ_JOIN(EnumTypeName, Count)> AZ_JOIN(EnumTypeName, Members) = \
-    {{ \
+    constexpr auto AZ_JOIN(EnumTypeName, Members) = AZStd::to_array<AZ_JOIN(EnumTypeName, EnumeratorValueAndString)> \
+    ({ \
         AZ_FOR_EACH_BIND1ST(INIT_ENUM_STRING_PAIR, EnumTypeName, __VA_ARGS__) \
-    }}; \
```

Comment:
The EnumTypeCount was based on the number of comma separated entries supplied for enum options.
So an additional count entry was being added for the trailing comma.

Instead, the AZStd::to_array function is used to deduce the number of entries that is added to the array of "enum option" -> "stringified enum option" pairs, and then the size of the array is used to initialize the count value